packages/zipkin-core/src/main/utils.ts

```diff
@@ -47,7 +47,7 @@ function is128bit(traceId: TraceId): boolean {
 export function serializeLinkerdHeader(traceId: TraceId): string {
     const serialized: ByteBuffer = ByteBuffer.concat([
         ByteBuffer.fromHex(traceId.spanId),
-        ByteBuffer.fromHex(traceId.parentId),
+        ByteBuffer.fromHex(traceId.parentSpanId.getOrElse('')),
```

Comment:
I think I don't understand the getOrElse() API. It seems like it makes things hard to use. Can parentSpanId ever actually be unset? Is it really safe to use an empty string in those cases?

I'm obviously not familiar with any of the code in this file.

Reply:
Yeah, the `zipkin` library updates (used for request tracing, it seems) changed the name but also wrapped these fields in some kind of `Optional` object where you have to call this method or something similar to unwrap the actual value (it's similar to other Java APIs I've seen in the past for wrapping possible `null` values). You can call `getOrElse()` without a default value and it would be `string | undefined`, but the `fromHex` function requires a `string`, so this seemed like the closest equivalent to how it would have been before? All the tests pass and this seems like the best equivalent - apparently `parentSpanId` can be unset (otherwise it wouldn't be this `Optional` type), and I believe an empty string is pretty safe to use (certainly more so than a `null` or `undefined` value).

I'm not crazy about the syntax either (obviously a `??` would seem a bit nicer than calling `getOrElse`) but I also don't know how to really get that changed other than switching it to a different library. 🤷